sharding/collation_test.go

```diff
@@ -129,6 +129,25 @@ func makeTxWithGasLimit(gl uint64) *types.Transaction {
 	return types.NewTransaction(0 /*nonce*/, common.HexToAddress("0x0") /*to*/, nil /*amount*/, gl, nil /*gasPrice*/, nil /*data*/)
 }
 
+func Test_CalculatePOC(t *testing.T) {
+	header := NewCollationHeader(big.NewInt(1), nil, big.NewInt(1), nil, []byte{})
+	body := []byte{0x56, 0xff}
+	transactions := []*types.Transaction{
+		makeTxWithGasLimit(0),
+		makeTxWithGasLimit(5),
+		makeTxWithGasLimit(20),
+		makeTxWithGasLimit(100),
+	}
+	c := NewCollation(header, body, transactions)
+
+	salt := []byte{1, 0x9f}
+	c.CalculateChunkRoot()
+	poc := c.CalculatePOC(salt)
+	if poc == *c.header.data.ChunkRoot {
+		t.Errorf("Salt: %x\nPoC: %x\nChunkroot: %x\n", salt, poc, c.header.data.ChunkRoot)
```

Comment:
Be a bit more clear with the error messages. Perhaps say Proof of Custody instead of PoC as the reader might not be as familiar with what it means.